### PR TITLE
FIX processing/rendering of attrsFormat

### LINKED ISSUE #2163: in POST v2/subscriptions the attrsFormat field is not stored correctly in mongoDB
in POST v2/subscriptions the `attrsFormat` notification field is not stored correctly in mongoDB

#### subscription curl
```
curl -i "http://localhost:1026/v2/subscriptions"  -H "Content-Type: application/json" -H "Fiware-ServicePath: /test" -d '{ "subject": {"entities": [{"idPattern": ".*"}], "condition": { "attrs": ["temperature"] }}, "notification": {"http": { "url": "http://localhost:1045/notify"}, "attrsFormat": "keyValues", "attrs": ["temperature"]}, "expires": "2017-04-05T14:00:00.00Z", "throttling": 0 }'
```

#### subscription response
```
HTTP/1.1 201 Created
Connection: Keep-Alive
Content-Length: 0
Location: /v2/subscriptions/573ed908038c02be6fe65039
Fiware-Correlator: 62e640e4-1e6d-11e6-b529-005056a20feb
Date: Fri, 20 May 2016 09:29:44 GMT
```

#### doc in mongo
```
{ "_id" : ObjectId("573ed908038c02be6fe65039"), "expiration" : NumberLong(1491400800), "reference" : "http://localhost:1045/notify", "extended" : false, "throttling" : NumberLong(0), "servicePath" : "/test", "status" : "active", "entities" : [ { "id" : ".*", "isPattern" : "true" } ], "attrs" : [ "temperature" ], "conditions" : [ { "type" : "ONCHANGE", "value" : [ "temperature" ] } ], "expression" : { "q" : "", "geometry" : "", "coords" : "", "georel" : "" }, "format" : "normalized" }
```

### PR DESCRIPTION
**Pull request by fgalan**
Fixes #2163 and #2133

### CODE REVIEW

File: src/lib/apiTypesV2/Subscription.cpp [hunk 1]
```diff
@@ -87,8 +87,12 @@ namespace ngsiv2
   /* ****************************************************************************
   *
   * Notification::toJson -
+  *
+  * FIXME P2: we should move 'attrsFormat' from Subject class to Notification
+  * class, to avoid passing attrsFormat as argument
+  *
```
Comment (kzangeli) [inline src/lib/apiTypesV2/Subscription.cpp]: Sounds like a good idea.
Why not in this very PR ?

File: test/functionalTest/cases/2163_bug_attrs_format_not_stored/bug_attrs_format_not_stored.test [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+        "notification": {
+                "http": {
+                        "url": "http://localhost:1045/notify"
+                },
+                "attrsFormat": "keyValues",
+                "attrs": ["temperature"]
+        },
+        "expires": "2017-04-05T14:00:00.00Z",
+        "throttling": 0
+}
```
Comment (kzangeli) [inline test/functionalTest/cases/2163_bug_attrs_format_not_stored/bug_attrs_format_not_stored.test]: Why so much indentation?

File: src/lib/apiTypesV2/Subscription.cpp (hunk 1 above)
Comment (fgalan) [inline src/lib/apiTypesV2/Subscription.cpp]: In order not to introduce noise in the template and blacklist functionality, that may be using affected parts in Subscriptions classes and corresponding parsing and rendering logic.

Thus, it is better to do it after template and blacklist functionality land into develop.

File: test/functionalTest/cases/2163_bug_attrs_format_not_stored/bug_attrs_format_not_stored.test (hunk 2 above)
Comment (fgalan) [inline test/functionalTest/cases/2163_bug_attrs_format_not_stored/bug_attrs_format_not_stored.test]: Copy-paste from JSONLint, maybe. Not very important.

File: src/lib/apiTypesV2/Subscription.cpp (hunk 1 above)
Comment (kzangeli) [inline src/lib/apiTypesV2/Subscription.cpp]: ok,

NTC

File: test/functionalTest/cases/2163_bug_attrs_format_not_stored/bug_attrs_format_not_stored.test (hunk 2 above)
Comment (kzangeli) [inline test/functionalTest/cases/2163_bug_attrs_format_not_stored/bug_attrs_format_not_stored.test]: Nope, 

NTC

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**